### PR TITLE
EPMRPP-38676 || Pattern analysis impl api

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/ws/converter/builders/PatternTemplateBuilder.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.function.Supplier;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class PatternTemplateBuilder implements Supplier<PatternTemplate> {
+
+	private PatternTemplate patternTemplate;
+
+	public PatternTemplateBuilder(CreatePatternTemplateRQ createPatternTemplateRQ) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/converter/builders/PatternTemplateBuilder.java]: move it to the method and create empty constructor.

File: src/main/java/com/epam/ta/reportportal/core/pattern/PatternAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	 * on {@link com.epam.ta.reportportal.entity.project.Project} with provided ID.
+	 * Every matched {@link com.epam.ta.reportportal.entity.pattern.PatternTemplate} will be attached
+	 * to the {@link com.epam.ta.reportportal.entity.item.TestItem}
+	 * using {@link com.epam.ta.reportportal.entity.pattern.PatternTemplateTestItem} relation
+	 *
+	 * @param launchId  {@link com.epam.ta.reportportal.entity.launch.Launch#id}, which {@link com.epam.ta.reportportal.entity.item.TestItem}
+	 *                  should be analyzed
+	 * @param projectId {@link com.epam.ta.reportportal.entity.project.Project#id}, which launch's items should be analyzed
+	 */
+	void analyzeTestItems(Long launchId, Long projectId);
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/pattern/PatternAnalyzer.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/pattern/impl/CreateRegexPatternTemplateHandler.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			throw new ReportPortalException(ErrorType.BAD_REQUEST_ERROR,
+					Suppliers.formattedSupplier("Provided regex pattern - '{}' is invalid", createPatternTemplateRQ.getValue()).get()
+			);
+		}
+
+		BusinessRule.expect(patternTemplateRepository.existsByProjectIdAndNameIgnoreCase(projectId, createPatternTemplateRQ.getName()),
+				equalTo(false)
+		).verify(ErrorType.RESOURCE_ALREADY_EXISTS, createPatternTemplateRQ.getName());
+
+		PatternTemplate patternTemplate = new PatternTemplateBuilder(createPatternTemplateRQ).withProjectId(projectId).get();
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/pattern/impl/CreateRegexPatternTemplateHandler.java]: done

File: src/main/java/com/epam/ta/reportportal/ws/converter/builders/PatternTemplateBuilder.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/ws/converter/builders/PatternTemplateBuilder.java]: done

File: src/main/java/com/epam/ta/reportportal/core/pattern/impl/PatternAnalyzerImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		this.issueTypeRepository = issueTypeRepository;
+		this.patternTemplateRepository = patternTemplateRepository;
+		this.patternAnalysisSelectorMapping = patternAnalysisSelectorMapping;
+		this.patternAnalysisTaskExecutor = patternAnalysisTaskExecutor;
+	}
+
+	@Override
+	public void analyzeTestItems(Long launchId, Long projectId) {
+
+		IssueType issueType = issueTypeRepository.findByLocator(TestItemIssueGroup.TO_INVESTIGATE.getLocator())
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/pattern/impl/PatternAnalyzerImpl.java]: fixed

**Pull request merged**